Dockerfile

```diff
@@ -3,9 +3,7 @@ ARG BASE_IMAGE=nvcr.io/nvidia/pytorch:24.02-py3
 FROM ${BASE_IMAGE} AS bionemo2-base
 
 # Install NeMo dependencies.
-WORKDIR /build
-
-ARG MAX_JOBS=4
+ARG MAX_JOBS=-1
```

Comment:
Do we need to cap the number of compilation workers? Is this to be a good citizen a CI system somewhere?

Reply:
It depends on available memory. If you pick too high MAX_JOBS the docker with our dependencies will not build without cache, ie at least it was happening in bionemo1 and TransformerEngine

Reply:
What does it mean --build-arg MAX_JOBS=N invalidates the cache?

Reply:
I suspect (at least this was happening locally) that if you pass in a MAX_JOBS argument via the command line, you'll only cache layers that are downstream of that particular configuration. So if I build locally with 
```docker buildx build . -t bionemo --build-arg MAX_JOBS=-1```
That won't cache those built layers if I later build and omit that MAX_JOBS argument.